src/views/studio/MyUploadsView/AssetLine/AssetLine.tsx

```diff
@@ -12,26 +17,116 @@ import {
   ProgressbarContainer,
 } from './AssetLine.style'
 import { UploadData } from '../AssetsGroupUploadBar/AssetsGroupUploadBar'
-import { Text, CircularProgressbar } from '@/shared/components'
-import { SvgAlertError, SvgAlertSuccess, SvgGlyphFileImage, SvgGlyphFileVideo } from '@/shared/icons'
+import { MessageDialog } from '@/components'
+import { Text, CircularProgressbar, Button } from '@/shared/components'
+import { SvgAlertError, SvgAlertSuccess, SvgGlyphFileImage, SvgGlyphFileVideo, SvgGlyphUpload } from '@/shared/icons'
+import { LiaisonJudgement } from '@/api/queries'
 
 type AssetLineProps = {
   isLast?: boolean
   asset: UploadData
 }
 
 const AssetLine: React.FC<AssetLineProps> = ({ isLast = false, asset }) => {
-  const isVideo = asset.type === 'video'
+  const navigate = useNavigate()
+  const {
+    activeUser: { channelId },
+  } = useActiveUser()
+  const { startFileUpload } = useUploadsManager(channelId || '')
+  const randomStorageProviderUrl = useRandomStorageProviderUrl()
+
+  const onDrop: DropzoneOptions['onDrop'] = useCallback(
+    async (acceptedFiles) => {
+      const [file] = acceptedFiles
+      const fileHash = await computeFileHash(file)
+      if (fileHash !== asset.ipfsContentId) {
+        setShowDialog(true)
+      } else {
+        startFileUpload(
+          file,
+          {
+            contentId: asset.contentId,
+            owner: asset.owner,
+            parentObject: {
+              type: asset.parentObject.type,
+              id: asset.parentObject.id,
+            },
+            type: asset.type,
+          },
+          randomStorageProviderUrl ?? '',
```

Comment:
not sure, but I think we shouldn't run `startFileUpload` if the `randomStorageProviderUrl` is `falsy`.